### ISSUE TITLE
Merge with upstream?

### ISSUE THREAD

**Issue by jordansissel** [awslabs/logstash-output-amazon_es#2]
I'd like to propose this repository be abandoned in favor of merging efforts with the Logstash elasticsearch output as well as the elasticsearch-ruby library.

The benefits here to authors is reduction in overall effort, and the benefit to users is less confusing interface, less feature chasing, and higher stability.

**Comment by malpani** [comment]
Yes, it is a good idea to integrate into elasticsearch-ruby and logstash-output-elasticsearch as it is also HTTP only now.

We are unsure on the time for our changes to get pulled in and wanted to have this available for Amazon Elasticsearch customers using request signing when we launched.  So plan is to continue supporting this for existing customers till the upstream merge is complete

The main differences from logstash-output-elasticsearch plugin are
* Signing support
* Client - custom timeouts and default port
* Absence of sniffing and other features which might not be relevant to Amazon Elasticsearch Service as requests get load balanced

Here is a proposed approach. 

Step 1
Faraday middleware for AWS V4 signing - separate module

Step 2
Add AWS Transport into elasticsearch-ruby which uses the rack middleware?
aws_transport - https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-transport/lib/elasticsearch/transport/transport/http

Step 3
logstash-output-elasticsearch
* new argument - hash of transport options?
transport_options => {transport: 'aws' aws_access_key_id: 'foo' aws_secret_key_id: 'bar' session_token: '' region: 'us-east-1' }
port => 80 (default is 9200)
Redefine client_options if transport chosen is aws - http://git.io/vcHnX

What are your thoughts?

**Comment by Asimov4** [comment]
@jordansissel how would you proceed? Just send a pull request in the 2 package that you suggested?

**Comment by fiunchinho** [comment]
any news on this?

**Comment by ScriptAutomate** [comment]
It's been almost a year now -- is this going to happen??

**Comment by malpani** [comment]
Ideally i would like to see this merged upstream as well. @jordansissel do you have any suggestions on the way forward.
